db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/add_provider.rb

```diff
@@ -1,60 +1,39 @@
-require 'net/ssh'
-
 DEPLOYMENT_TYPES = {
-  :origin            => "ManageIQ::Providers::Openshift::ContainerManager",
-  :enterprise        => "ManageIQ::Providers::OpenshiftEnterprise::ContainerManager",
-  :atomic            => "ManageIQ::Providers::Atomic::ContainerManager",
-  :atomic_enterprise => "ManageIQ::Providers::AtomicEnterprise::ContainerManager"
+  :origin               => "ManageIQ::Providers::Openshift::ContainerManager",
+  :openshift_enterprise => "ManageIQ::Providers::OpenshiftEnterprise::ContainerManager",
 }. freeze
 
 def provider_token
-  token = ""
-  Net::SSH.start($evm.root['deployment_master'], $evm.root['user'], :paranoid => false, :forward_agent => true,
-                 :key_data => $evm.root['private_key']) do |ssh|
-    cmd = "oc get -n management-infra sa/management-admin --template='{{range .secrets}}{{printf " + '"%s\n"' \
-          " .name}}{{end}}' | grep token"
-    token_key = ssh.exec!(cmd)
-    token_key.delete!("\n")
-    cmd = "oc get -n management-infra secrets #{token_key} --template='{{.data.token}}' | base64 -d"
-    token = ssh.exec!(cmd)
-  end
-  token
+  $evm.root['container_deployment'].perform_agent_commands($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), ["sudo oc sa get-token management-admin -n management-infra"])[:stdout]
```

Comment:
@alongoldboim it seems that `management-admin` and `management-infra` should **at least** become constants.